metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -214,21 +213,34 @@ void traverse_anno_chunked(
 
         before_chunk(block_size);
 
-        succ_chunk.resize(block_size);
-        pred_chunk.resize(0);
-        pred_chunk_idx.resize(block_size + 1);
-        pred_chunk_idx[0] = 0;
-
-        for (uint64_t i = 0; i < block_size; ++i) {
-            succ_chunk[i] = succ[chunk + i];
-            pred_chunk_idx[i + 1] = pred_chunk_idx[i];
-            while (*pred_boundary_it == 0) {
-                ++pred_chunk_idx[i + 1];
-                pred_chunk.push_back(*pred_it);
-                ++pred_it;
-                ++pred_boundary_it;
+        #pragma omp parallel
+        #pragma omp single
+        {
+            #pragma omp task
+            {
+                succ_chunk.resize(block_size);
+                for (uint64_t i = 0; i < block_size; ++i) {
+                    succ_chunk[i] = succ[chunk + i];
+                }
+            }
+
+            #pragma omp task
+            {
+                pred_chunk.resize(0);
+                pred_chunk_idx.resize(block_size + 1);
+                pred_chunk_idx[0] = 0;
+
+                for (uint64_t i = 0; i < block_size; ++i) {
+                    pred_chunk_idx[i + 1] = pred_chunk_idx[i];
+                    while (*pred_boundary_it == 0) {
+                        ++pred_chunk_idx[i + 1];
+                        pred_chunk.push_back(*pred_it);
+                        ++pred_it;
+                        ++pred_boundary_it;
+                    }
+                    ++pred_boundary_it;
+                }
```

Comment:
add a comment describing what this is supposed to be doing, it's a bit hard to follow

Reply:
Wouldn't it be easier to do this by first populating `pred_chunk_idx` by iterating through `pred_boundary_it`, then doing the copy to `pred_chunk` in a single call?

Reply:
to make this a bit faster, does `next_bit` work with `int_vector_buffer<1>`?

Reply:
though in the end, reading from the file will be the bottleneck

Reply:
added comments